### DISCUSSION TITLE
Migration trouble

### DISCUSSION THREAD

**Discussion by Hassenhajsalem** [ConsenSys-archive/truffle#5921] [General Q&A]
# truffle migrate
This version of µWS is not compatible with your Node.js build:

Error: Cannot find module './uws_linux_x64_111.node'
Falling back to a NodeJS implementation; performance may be degraded.

Error: The network id specified in the truffle config (5777) does not match the one returned by the network (1677063514397).  Ensure that both the network and the provider are properly configured.
    at Object.detectAndSetNetworkId (/usr/local/lib/node_modules/truffle/build/webpack:/packages/environment/environment.js:114:1)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at Object.detect (/usr/local/lib/node_modules/truffle/build/webpack:/packages/environment/environment.js:24:1)
    at Object.module.exports [as run] (/usr/local/lib/node_modules/truffle/build/webpack:/packages/core/lib/commands/console/run.js:21:1)
    at runCommand (/usr/local/lib/node_modules/truffle/build/webpack:/packages/core/lib/command-utils.js:297:1)
Truffle v5.7.7 (core: 5.7.7)
Node v19.6.1

**Comment by LoveKosmos** [comment]
Hi, it is recommended to use Latest LTS Version of Node, which is 18.14.2 (includes npm 9.5.0).

Could you give it a try?

**Reply by Hassenhajsalem** [reply]
Thanks , i resolve the problem by modify truffle-config.js and change the network id

**Reply by vishalrajput9560** [reply]
Hi 
I am still getting the same error I tried node all 14-18 versions.  But did  not get any solution
please help

**Reply by hassenhajsalem1** [reply]
can you share a screenshot for the error?

**Reply by vishalrajput9560** [reply]
![error](https://user-images.githubusercontent.com/46858313/226097103-bfd70314-47f6-4d1e-a102-baccd93b08cc.png)

**Reply by vishalrajput9560** [reply]
here it is

**Reply by hassenhajsalem1** [reply]
try this command in the cmd : truffle migrate --network development

**Reply by mr-possible** [reply]
@vishalrajput9560 Few things:

1. Uninstall Node completely and only install the LTS version.
2. Avoid the Truffle VScode extension for now, and remove it if you're using it. It forces us to install the latest version of Node.
3. Try using Solidity Version 0.8.18 - I think it is far more stable than any other version to date.
4. Make sure to use the latest `web3.js` dependency. That is mostly the issue at times.
5. Under `migrations` folder, create a file called as `1_initial_migration.js` and inside it, do like what is given below:
```
const MyContract = artifacts.require("<your contract name comes here>");

module.exports = function (deployer) {
    deployer.deploy(MyContract, "<any arguments if your constructor needs it>");
};
```
6. Make sure you create a folder called `truffle-artifacts` in the project's root directory. This will force Truffle to store the compiled contracts in an isolated location so that your project's build folder (if any) doesn't interfere with that.
7. Keep your `truffle-config.js` file like the one given below:

```
module.exports = {
  networks: {
    development: {
      host: "127.0.0.1",     // Localhost (default: none)
      port: 7545,            // Standard Ethereum port (default: none)
      network_id: "*",       // Any network (default: none)
    }
  },
  // Set default mocha options here, use special reporters, etc.
  mocha: {
    // timeout: 100000
  },
  
  contracts_build_directory: "./truffle-artificats",
  
  // Configure your compilers
  compilers: {
    solc: {
      version: "0.8.18"
    },
  }
}
```

Hope it helps. Let us know. Cheers mate!

**Comment by vishalrajput9560** [comment]
![image](https://user-images.githubusercontent.com/46858313/226097293-b048116f-e5c3-4bee-84eb-1fa0820eda1d.png)

**Comment by vishalrajput9560** [comment]
![image](https://user-images.githubusercontent.com/46858313/226098933-000633e1-c699-4e40-886e-0a16a394d260.png)

**Reply by vishalrajput9560** [reply]
not working

**Reply by Hassenhajsalem** [reply]
>>vi 1_deploy_contract.js    //Add a configuration file under the migrations directory

Example: 

 const FirstContract = artifacts.require("FirstContract");  

module.exports = function(deployer) { deployer.deploy(FirstContract); };
